tests/php/ModelAdminTest.php

```diff
@@ -126,4 +129,84 @@ public function testGetGridField()
             'The import button is still there'
         );
     }
+
+    public function testGetManagedModels()
+    {
+        $admin = new ModelAdminTest\MultiModelAdmin();
+
+        $models = $admin->getManagedModels();
+
+        $this->assertEquals(
+            [
+                'dataClass' => Contact::class,
+                'title' => 'Contacts'
+            ],
+            $models[Contact::class],
+            'Managed Model that only provide a string get nomalised'
+        );
+
+        $this->assertEquals(
+            [
+                'dataClass' => Player::class,
+                'title' => 'Ice Hockey Players'
+            ],
+            $models['Player'],
+            'Managed Model with an explicit dataClass can have a custom URL key'
+        );
+
+        $this->assertEquals(
+            [
+                'dataClass' => Player::class,
+                'title' => 'Rugby Players'
+             ],
+            $models[Player::class],
+            'Managed Model default to using the array key when a dataClass is not provided'
+        );
+    }
+
+    public function testGetManagedModelTabs()
+    {
+        $mock = $this->getMockBuilder(ModelAdminTest\MultiModelAdmin::class)
```

Comment:
This seems wrong to me that we're using a mock for this given there's a test class explicitly for this which we can easily instantiate.  It seems like the only reason that it's used is to be able to access the protected method `getManagedModelTabs()`.

I particularly don't like that we're coercing the return value of `getManagedModels()`

Why not just 
```
$admin = new ModelAdminTest\MultiModelAdmin();
$tabs = $admin->getManagedModelTabs()->toNestedArray();
```

?

Reply:
`getManagedModelTabs` calls `getManagedModels` internally. The original PR that introduced the bug updated `getManagedModels` so it always uses the `'Player' => ['dataClass' => Player::class, 'title' => 'Ice Hockey Players']` format.

I need to make sure `getManagedModelTabs` doesn't fall over when `getManagedModels` returns the legacy format. The only way I could do that is:
- mock the function like I'm doing right here
- create yet another ModelAdmin test class and override `getManagedModels`

I think the mock function has the advantage of showing exactly the data input we are testing next to the assertion.

Reply:
OK, in that case I'd recommend adding an inline comment explain the mock is explicitly coercing the return value of `getManagedModels()` to match the legacy format

May as well also remove `->expects($this->once())` since the purpose of the mock is not for assertion, it's specially for data generation

Reply:
Added a comment. The `once` call is already there.